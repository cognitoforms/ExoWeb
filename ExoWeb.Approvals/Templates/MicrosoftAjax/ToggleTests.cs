using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace ExoWeb.Approvals.Templates.MicrosoftAjax
{
	[TestClass]
	[UseReporter(typeof(DiffReporter))]
	public class ToggleTests : BaseRenderTests
	{
		[TestMethod]
		public void Toggle_Show()
		{
			var outputMarkup = Render(
				@"<div class=""sys-template"" sys:attach=""dataview"" dataview:data=""{~ context.model.request, source=window }"">
					<div sys:attach=""toggle"" toggle:on=""{binding Description}"">
						<span>{binding Description}</span>
					</div>
				</div>");

			ApprovalTests.Approvals.VerifyHtml(outputMarkup);
		}

		[TestMethod]
		public void Toggle_ShowWithInlineDisplayStyle()
		{
			var outputMarkup = Render(@"
				<div class=""sys-template"" sys:attach=""dataview"" dataview:data=""{~ context.model.request, source=window }"">
					<div style=""display:inline;"" sys:attach=""toggle"" toggle:on=""{binding Description}"">
						<span>{binding Description}</span>
					</div>
				</div>");

			ApprovalTests.Approvals.VerifyHtml(outputMarkup);
		}
	}
}
