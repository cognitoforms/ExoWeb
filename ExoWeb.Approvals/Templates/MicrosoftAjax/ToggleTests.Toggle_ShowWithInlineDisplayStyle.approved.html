<output xmlns:sys="javascript:Sys" xmlns:dataview="javascript:Sys.UI.DataView" xmlns:content="javascript:ExoWeb.UI.Content" xmlns:template="javascript:ExoWeb.UI.Template" xmlns:toggle="javascript:ExoWeb.UI.Toggle" xmlns:behavior="javascript:ExoWeb.UI.Behavior" xmlns:html="javascript:ExoWeb.UI.Html" xmlns:togglegroup="javascript:ExoWeb.UI.ToggleGroup">
  <div class="sys-template" id="exo0">
    <div style="display:inline;" sys:attach="toggle" toggle:on="{binding Description}">
      <span>{binding Description}</span>
    </div>
  </div>
  <div data-sys-attach="dataview" data-dataview-data="{~ context.model.request, source=window }" data-sys-tmplidx="0" id="exo1" class="sys-ignore">
    <!--item:s1-->
    <div style="display:inline;" data-sys-attach="toggle" data-toggle-on="{binding Description}" class="toggle-on">
      <span data-sys-innerhtml="{binding Description}">Something doesn't work</span>
    </div>
    <!--/item-->
  </div>
  <script type="text/javascript">$exoweb({ domReady: function() { Sys.Application.linkElement(document.getElementById("exo1"), document.getElementById("exo0")); } });</script>
</output>